src/cpp/rtps/reader/StatelessReader.cpp

```diff
@@ -739,6 +739,7 @@ bool StatelessReader::processDataFragMsg(
                         {
                             work_change->copy_not_memcpy(change_to_add);
                             work_change->serializedPayload.length = sampleSize;
+                            work_change->instanceHandle = c_InstanceHandle_Unknown;
```

Comment:
This should also be applied after the previous `copy_not_memcpy` call in this same function
```suggestion
                            work_change->instanceHandle.clear();
```